docs/communicate-with-javascript.md

```diff
@@ -3154,3 +3165,622 @@ let default = 10;
+```
+
+See the [Using global functions or values](#using-global-functions-or-values)
+section for more information.
+
+#### `window?`: does global variable exist
+
+```ocaml
+let _ = match [%mel.external window] with
+| Some _ -> "window exists"
+| None -> "window does not exist"
+```
+```reasonml
+let _ =
+  switch ([%mel.external window]) {
+  | Some(_) => "window exists"
+  | None => "window does not exist"
+  };
+```
+
+See the [Detect global variables](#detect-global-variables) section for more
+information.
+
+#### `Math.PI`: variable in global module
+
+```ocaml
+external pi : float = "PI" [@@mel.scope "Math"]
+```
+```reasonml
+[@mel.scope "Math"] external pi: float = "PI";
+```
+
+See the [Binding to properties inside a module or
+global](#binding-to-properties-inside-a-module-or-global) section for more
+information.
+
+#### `console.log`: function in global module
+
+```ocaml
+external log : 'a -> unit = "log" [@@mel.scope "console"]
+```
+```reasonml
+[@mel.scope "console"] external log: 'a => unit = "log";
+```
+
+See the [Binding to properties inside a module or
+global](#binding-to-properties-inside-a-module-or-global) section for more
+information.
+
+### Modules
+
+#### `const path = require('path'); path.join('a', 'b')`: function in CommonJS/ES6 module
+
+```ocaml
+external join : string -> string -> string = "join" [@@mel.module "path"]
+let dir = join "a" "b"
+```
+```reasonml
+[@mel.module "path"] external join: (string, string) => string = "join";
+let dir = join("a", "b");
+```
+
+See the [Using functions from other JavaScript
+modules](#using-functions-from-other-javascript-modules) section for more
+information.
+
+#### `const foo = require('foo'); foo(1)`: import entire module as a value
+
+```ocaml
+external foo : int -> unit = "foo" [@@mel.module]
+let () = foo 1
+```
+```reasonml
+[@mel.module] external foo: int => unit = "foo";
+let () = foo(1);
+```
+
+See the [Using functions from other JavaScript
+modules](#using-functions-from-other-javascript-modules) section for more
+information.
+
+#### `import foo from 'foo'; foo(1)`: import ES6 module default export
+
+```ocaml
+external foo : int -> unit = "default" [@@mel.module "foo"]
+let () = foo 1
+```
+```reasonml
+[@mel.module "foo"] external foo: int => unit = "default";
+let () = foo(1);
+```
+
+See the [Using functions from other JavaScript
+modules](#using-functions-from-other-javascript-modules) section for more
+information.
+
+#### `const foo = require('foo'); foo.bar.baz()`: function scoped inside an object in a module
+
+```ocaml
+module Foo = struct
+  module Bar = struct
+    external baz : unit -> unit = "baz" [@@mel.module "foo"] [@@mel.scope "bar"]
+  end
+end
+
+let () = Foo.Bar.baz ()
+```
+```reasonml
+module Foo = {
+  module Bar = {
+    [@mel.module "foo"] [@mel.scope "bar"] external baz: unit => unit = "baz";
+  };
+};
+
+let () = Foo.Bar.baz();
+```
+
+It is not necessary to nest the binding inside Reason modules, but mirroring the
```

Comment:
```suggestion
It is not necessary to nest the binding inside Reason/OCaml modules, but mirroring the
```

This wording could cause confusion to the reader that there is a differentiation between Reason and OCaml modules? We could also just say 'OCaml' modules?

Reply:
Good point, yeah let's use OCaml modules.